tests/test_utils_http.py

```diff
@@ -311,3 +312,23 @@ def _is_uuid(string: str) -> bool:
 )
 def test_fix_hf_endpoint_in_url(base_url: str, endpoint: Optional[str], expected_url: str) -> None:
     assert fix_hf_endpoint_in_url(base_url, endpoint) == expected_url
+
+
+def test_adjust_range_header():
+    # Basic cases
+    assert _adjust_range_header(None, 10) == "bytes=10-"
+    assert _adjust_range_header("bytes=0-100", 10) == "bytes=10-100"
+    assert _adjust_range_header("bytes=-100", 10) == "bytes=-90"
+
+    with pytest.raises(RuntimeError):
+        _adjust_range_header("invalid", 10)
+
+    with pytest.raises(RuntimeError):
+        _adjust_range_header("bytes=-", 10)
+
+    # Multiple ranges
+    assert _adjust_range_header("bytes=0-100,200-300", 10) == "bytes=10-"
+
+    # Resume size exceeds range
+    assert _adjust_range_header("bytes=0-100", 150) is None
+    assert _adjust_range_header("bytes=-50", 100) is None
```

Comment:
Same here, better to raise rather than returning None i.e. better not to support rather than downloading everything

Reply:
pushed